File: services/chess_service/src/main/java/net/chess_platform/chess_service/chess/piece/Bishop.java
package net.chess_platform.chess_service.chess.piece;

import java.util.List;

import net.chess_platform.chess_service.chess.Chessboard;
import net.chess_platform.chess_service.chess.move.Move;
import net.chess_platform.chess_service.chess.piece.behavior.BishopBehavior;
import net.chess_platform.chess_service.chess.piece.behavior.PieceBehavior;

public class Bishop extends AbstractPiece {

    private final PieceBehavior bishopMoves = new BishopBehavior();

    public Bishop(Color color) {
        super(color, Type.BISHOP);

    }

    @Override
    public List<Move> getMoves(Chessboard board, int row, int col) {
        return bishopMoves.getMoves(board, getType(), getColor(), row, col);
    }

    @Override
    public String toString() {
        return getColor() == Color.WHITE ? Character.toString(9815) : Character.toString(9821);
    }
}

File: app/src/chess/pieces/Bishop.ts
import { Chessboard } from "../Chessboard";
import { Move } from "../moves/Move";
import { AbstractPiece } from "./AbstractPiece";
import { BishopBehavior } from "./behavior/BishopBehavior";
import { Color } from "./Piece";

export class Bishop extends AbstractPiece {

    private behavior = new BishopBehavior();

    constructor(color: Color, moveCount?: number) {
        super(color, "BISHOP", moveCount);
    }

    getLegalMovesImpl(board: Chessboard, row: number, col: number): Move[] {
        return this.behavior.getMoves(board, this.getType(), this.getColor(), row, col);
    }

    override toString(): string {
        return this.getColor() == "WHITE" ? String.fromCodePoint(9815) : String.fromCodePoint(9821);
    }
}
